File: src/main/java/me/nasukhov/TukitaLearner/study/Fact.java
package me.nasukhov.TukitaLearner.study;

import jakarta.persistence.*;

@Entity
@Table(name = "facts")
public class Fact {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String text;

    protected Fact() {
    }

    public Fact(String text) {
        this.text = text;
    }

    public Long getId() {
        return id;
    }

    public String getText() {
        return text;
    }
}

File: src/main/kotlin/me/nasukhov/tukitalearner/study/Fact.kt
package me.nasukhov.tukitalearner.study

import jakarta.persistence.*

@Entity
@Table(name = "facts")
class Fact {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0
        private set

    @Column(columnDefinition = "TEXT", nullable = false)
    lateinit var text: String
        private set

    private constructor()

    constructor(text: String) {
        this.text = text
    }
}
